### PR TITLE
Generalize degradedTests execution for local and CF

### PR DESCRIPTION
**Pull request by RoopGuron**
Signed-off-by: Henry <henry.zhao1@ge.com>

### CODE REVIEW

File: scripts/setupDegradedTests.sh [hunk 1]
```diff
@@ -16,7 +32,7 @@ uaac curl -X POST /identity-zones -H 'Content-Type: application/json' -d'{ "id":
 uaac -t curl -H "X-Identity-Zone-Id:test-saml-zone" -XPOST -H"Content-Type:application/json" -H"Accept:application/json" --data '{ "client_id" : "admin", "client_secret" : "adminsecret", "scope" : ["uaa.none"], "resource_ids" : ["none"], "authorities" : ["uaa.admin","clients.read","clients.write","clients.secret","scim.read","scim.write","clients.admin", "sps.write", "sps.read", "zones.test-saml-zone.admin", "idps.read", "idps.write"], "authorized_grant_types" : ["client_credentials"]}' /oauth/clients
 
 #Login to test-saml-zone
-uaac target http://test-saml-zone.localhost:8080/uaa
+uaac target ${PROTOCOL}://test-saml-zone.$PUBLISHED_DOMAIN
```
Comment (Bitty9) [inline scripts/setupDegradedTests.sh]: When creating admin clients for zones, parameterize the secrets and set them in uaa-cf-release.

File: scripts/setupDegradedTests.sh [hunk 2]
```diff
@@ -25,12 +41,14 @@ SAML_IDP_METADATA_RAW=$(echo $SAML_IDP_RESPONSE | sed s/.*RESPONSE\ BODY://)
...
+uaac user add samluser1 -p samluser1 --email samluser1@ge.com || true
+
+uaac user add samluser2 -p samluser2 --email samluser2@ge.com || true
+
+uaac user add 1234 -p user3 --email user3@ge.com || true
 
 #Login to test-platform-zone
-uaac target http://test-platform-zone.localhost:8080/uaa
+uaac target ${PROTOCOL}://test-platform-zone.$PUBLISHED_DOMAIN
 uaac token client get admin -s adminsecret
```
Comment (Bitty9) [inline scripts/setupDegradedTests.sh]: This admin secret should be parameterized.

File: scripts/setupDegradedTests.sh [hunk 3]
```diff
@@ -128,16 +134,18 @@ uaac curl /saml/service-providers -XPOST -H 'Content-Type: application/json' -d
 }'
 
 #Login to test-app-zone
-uaac target http://test-app-zone.localhost:8080/uaa
+uaac target ${PROTOCOL}://test-app-zone.$PUBLISHED_DOMAIN
 uaac token client get admin -s adminsecret
```
Comment (Bitty9) [inline scripts/setupDegradedTests.sh]: This admin secret should be parameterized.

File: server/src/main/java/org/cloudfoundry/identity/uaa/mode/degraded/DegradedModeUaaFilter.java [hunk 4]
```diff
@@ -37,7 +37,8 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
                 requestPath.startsWith("/saml")) {
             filterChain.doFilter(request, response);
         }else {
-            String responseToClient = "{\"error\": \"UAA currently in degraded mode, operation not permitted.\"}";
+            String responseToClient = "{\"error\": \"UAA currently in read-only mode, operation not permitted. " +
+                    "UAA can still issue tokens and respond to queries\"}";
             logger.warn("Operation Not permitted in degraded mode::"+uri.substring(contextPath.length()));
```
Comment (Bitty9) [inline server/src/main/java/org/cloudfoundry/identity/uaa/mode/degraded/DegradedModeUaaFilter.java]: Were we going to add the full list of functional operations?

File: uaa/src/test/java/org/cloudfoundry/identity/uaa/degraded/DegradedSamlLoginTests.java [hunk 5]
```diff
@@ -61,23 +60,45 @@
...
+    private String testRedirectUri;
     private String zoneAdminToken;
+    private String baseBasicAuth;
 
 
     @Before
     public void setup() throws Exception {
-        baseUrl = "http://" + zoneSubdomain + ".localhost:8080/uaa";
+        System.out.println("Environment"+environment);
+        baseUaaZoneHost = environment.getProperty("runAgainstCloud") != null ? (publishedHost + "." + cfDomain) : "localhost:8080/uaa";
```
Comment (Bitty9) [inline uaa/src/test/java/org/cloudfoundry/identity/uaa/degraded/DegradedSamlLoginTests.java]: The variable name "runAgainstCloud". Same as the other one... RUN_AGAINST_CLOUD.

File: uaa/src/test/java/org/cloudfoundry/identity/uaa/degraded/DegradedSamlLoginTests.java [hunk 6]
```diff
@@ -101,17 +122,15 @@ public void testGetTokenKey() throws Exception {
...
-                "http://localhost:8080/uaa" + "/token_key",
+                protocol + baseUaaZoneHost + "/token_key",
                 HttpMethod.GET,
                 getHeaders,
                 Map.class
         );
         assertEquals(HttpStatus.OK, tokenKeyGet.getStatusCode());
-        assertEquals("MAC", tokenKeyGet.getBody().get("kty"));
-        assertEquals("HS256", tokenKeyGet.getBody().get("alg"));
     }
```
Comment (Bitty9) [inline uaa/src/test/java/org/cloudfoundry/identity/uaa/degraded/DegradedSamlLoginTests.java]: Why are these asserts removed?

File: uaa/src/test/java/org/cloudfoundry/identity/uaa/degraded/DegradedSamlLoginTests.java [hunk 7]
```diff
@@ -297,8 +317,11 @@ private String getRequestUrlFromHarLogEntry(LogEntry logEntry)
             throws IOException {
 
         Map<String, Object> message = JsonUtils.readValue(logEntry.getMessage(), new TypeReference<Map<String,Object>>() {});
+        System.out.println("message:"+ message);
         Map<String, Object> log = (Map<String, Object>) message.get("log");
```
Comment (Bitty9) [inline uaa/src/test/java/org/cloudfoundry/identity/uaa/degraded/DegradedSamlLoginTests.java]: Remove all System.out statements. Make log statements if you think they are necessary.

File: uaa/src/test/java/org/cloudfoundry/identity/uaa/degraded/DegradedSamlLoginTests.java (hunk 6 above)
Comment (RoopGuron) [inline uaa/src/test/java/org/cloudfoundry/identity/uaa/degraded/DegradedSamlLoginTests.java]: These are distinct for local uaa and ge uaa. 
The tokenKey is successfully retrieved, these were additional asserts.

**Pull request merged**